pkg/pool-stable-phantom/contracts/StablePhantomPool.sol

```diff
@@ -861,6 +928,22 @@ contract StablePhantomPool is IRateProvider, BaseGeneralPool, ProtocolFeeCache {
         return index < _bptIndex ? index : index.sub(1);
     }
 
+    /**
+     * @notice Returns an array with the `bptIndex` value removed. Also returns the virtual supply,
+     * if called with current token balances.
+     *
+     * @dev This function can be used in two ways. It will always drop the BPT item from the
+     * `amounts` input array, and return the result in `amountsWithoutBpt`.
+     *
+     * If the amounts given are token balances (e.g., from join/exit hooks), then the
+     * `virtualSupply` will be accurate, and is returned as a convenience, since often both
+     * values are needed together.
+     *
+     * It can be called with other kinds of amounts, but in that case the value returned in
+     * virtualSupply will not be meaningful.
+     *
+     * You can always call `getVirtualSupply` directly, if that is what you need.
+     */
     function _dropBptItem(uint256[] memory amounts)
```

Comment:
These docs are great, thanks. I'd suggest on a future PR to split this function into two, one that computes virtual supply and one that does not (since the virtual supply computation is not cheap, and we often don't use it).

Unfortunately the language does not support overloads that differ solely on the return type, so we'll need to have two different names (e.g. `_dropBptItem` and `_dropBptItemWithSupply` or w/e).